build.go

```diff
@@ -38,6 +50,36 @@ func Build(projectPathParser PathParser, buildManager BuildManager, clock chrono
 		}
 		nodeCacheLayer = setLayerFlags(nodeCacheLayer, context.Plan.Entries)
 
+		var globalNpmrcPath string
+		bindings, err := bindingResolver.Resolve("npmrc", "", context.Platform.Path)
+		if err != nil {
+			return packit.BuildResult{}, err
+		}
+
+		if len(bindings) > 1 {
+			return packit.BuildResult{}, errors.New("binding resolver found more than one binding of type 'npmrc'")
+		}
+
+		if len(bindings) == 1 {
+			logger.Process("Loading npmrc service binding")
+
+			npmrcExists := false
+			for key := range bindings[0].Entries {
+				if key == ".npmrc" {
+					npmrcExists = true
+					break
+				}
+			}
```

Comment:
Should use `servicebindings.ResolveOne()` instead of this logic.

Reply:
Unfortunately, `servicebindings.ResolveOne()` fails no binding of the provided type is found. We can't use it here since the `npmrc` binding is optional

Reply:
Could you do this?

```go
serviceBindingResolver := servicebindings.NewResolver()
npmrc, err := serviceBindingResolver.ResolveOne("npmrc", "", context.Platform.Path)

// err == nil ONLY if binding exists and there was no IO error
if err == nil {
	logger.Process("Using .npmrc binding")
	npmrcPath = filepath.Join(npmrc.Path, ".npmrc")
}
```